odl/operator/oputils.py

```diff
@@ -64,66 +69,33 @@ def matrix_representation(op):
 
```

Comment:
Commenting here: Currently our example is much like a basic unit test and doesn't really educate users a lot. Maybe we can turn this into a more interesting example, for instance with a `Gradient` operator.
And then, now that the returned object is not necessarily a matrix any more, it would be good to show how to do the equivalent thing to `op(x)` with the matrix using `np.tensordot`.
Here's a suggestion:
```python
>>> space = odl.uniform_discr([0, 0], [2, 2], (2, 2))
>>> grad = odl.Gradient(space, pad_mode='symmetric')
>>> odl.matrix_representation(grad)  # this may be a bit long
array([[[[[-1.,  0.],
          [ 1.,  0.]],

         [[ 0., -1.],
          [ 0.,  1.]]],


        [[[ 0.,  0.],
          [ 0.,  0.]],

         [[ 0.,  0.],
          [ 0.,  0.]]]],



       [[[[-1.,  1.],
          [ 0.,  0.]],

         [[ 0.,  0.],
          [ 0.,  0.]]],


        [[[ 0.,  0.],
          [-1.,  1.]],

         [[ 0.,  0.],
          [ 0.,  0.]]]]])
>>> space = odl.uniform_discr([0, 0, 0], [2, 3, 4], (2, 3, 4))
>>> lapl = odl.Laplacian(space)
>>> lapl_mat = odl.matrix_representation(lapl)
>>> x = space.element(lambda x: x[0] ** 2 + 2 * x[1] ** 2 - x[2] ** 3)  # or sth else for less weird numbers
>>> lapl(x)
uniform_discr([ 0.,  0.,  0.], [ 2.,  3.,  4.], (2, 3, 4)).element(
    [[[   0.875,    2.25 ,   20.75 ,  159.625],
      [  -6.5  ,   -4.375,    1.875,  109.5  ],
      [ -47.125,  -33.75 ,  -15.25 ,  111.625]],
    
     [[  -9.125,   -5.75 ,   12.75 ,  149.625],
      [ -14.5  ,  -10.375,   -4.125,  101.5  ],
      [ -57.125,  -41.75 ,  -23.25 ,  101.625]]]
)
>>> np.tensordot(lapl_mat, x, axes=3)
array([[[   0.875,    2.25 ,   20.75 ,  159.625],
        [  -6.5  ,   -4.375,    1.875,  109.5  ],
        [ -47.125,  -33.75 ,  -15.25 ,  111.625]],

       [[  -9.125,   -5.75 ,   12.75 ,  149.625],
        [ -14.5  ,  -10.375,   -4.125,  101.5  ],
        [ -57.125,  -41.75 ,  -23.25 ,  101.625]]])
```